addons/api/addon/workers/sqlite-worker.js

```diff
@@ -37,21 +38,32 @@ const methods = {
       method: 'getDatabaseName',
     });
 
+    function setupDb() {
+      const [row] = db.exec('PRAGMA user_version;', { rowMode: 'object' });
+      // Check if we're on a newer schema version and clear the database if so
+      if (SCHEMA_VERSION > row.user_version) {
+        db.exec(CLEAR_DB);
+      }
+
+      db.exec(CREATE_TABLES(SCHEMA_VERSION));
+      // Force an analyze on initial connection
+      db.exec('PRAGMA optimize=0x10002;');
+    }
+
+    // Instantiate an in memory DB if we're not in a secure context as OPFS is unavailable otherwise
+    if (!isSecure) {
+      db = new sqlite3.oo1.DB(':memory:');
```

Comment:
None blocking comment and only asking for leraning purposes: Did we test/try this with a big amount of data? I am wondering the memory consumption with a big load, I am sure shouldn't be a problem, specially after the changes you made to batch the digestion of big amounts.